bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java

```diff
@@ -139,14 +139,34 @@ public static BError createError(Throwable error) {
         return createError(StringUtils.fromString(error.getMessage()));
     }
 
+    /**
+     * Create a error value with given error type  defined in the given module.
+     *
+     * @param module        module name.
+     * @param errorTypeName error type name
+     * @param message       error message
+     * @param cause         error cause
+     * @param details       error details
+     * @return error value
+     * @throws BError if given error type is not defined in the ballerina module.
+     */
+    public static BError createError(Module module, String errorTypeName,
+                                     BString message, BError cause, Object details) {
+        io.ballerina.runtime.internal.values.ValueCreator
+                valueCreator = io.ballerina.runtime.internal.values.ValueCreator.getValueCreator(module.toString());
+        return valueCreator.createErrorValue(errorTypeName, message, cause, details);
+    }
+
     /**
      * Create an distinct error with given typeID, typeIdPkg and reason.
      *
      * @param typeIdName type id
      * @param typeIdPkg  type id module
      * @param message  error message
      * @return new error
+     * @deprecated Use @createError(module, errorTypeName, message, cause, details) to create a distinct error.
```

Comment:
Previously we were able to create errors without defining the type in the bal. We actually create new type every time which is a bug. And this is also not similar to create a record or object, where we need to define a type in the bal. I will remove these once stdlib usages  are migrated.